src/storage/generic.rs

```diff
@@ -0,0 +1,19 @@
+use tokio::sync::RwLock;
+
+#[derive(Default)]
+pub struct Generic {
+    state: RwLock<Option<super::StoredProperties>>,
+}
+
+impl super::Storage for Generic {
+    type Error = std::convert::Infallible;
+
+    async fn load(&self) -> Result<Option<super::StoredProperties>, Self::Error> {
+        Ok((*self.state.read().await).clone())
+    }
+
+    async fn store(&mut self, properties: &super::StoredProperties) -> Result<(), Self::Error> {
```

Comment:
Given that `store` here takes an `&mut self`, I don't think we need any locking. I removed the inner `RwLock` locally and `cargo check` didn't complain.